hooks.go

```diff
@@ -36,17 +43,146 @@ type Hooks struct {
 	onGroup        []OnGroupHandler
 	onGroupName    []OnGroupNameHandler
 	onListen       []OnListenHandler
+	onPreStartup   []OnPreStartupMessageHandler
+	onPostStartup  []OnPostStartupMessageHandler
 	onPreShutdown  []OnPreShutdownHandler
 	onPostShutdown []OnPostShutdownHandler
 	onFork         []OnForkHandler
 	onMount        []OnMountHandler
 }
 
-// ListenData is a struct to use it with OnListenHandler
+// startupMessageEntry represents a single line of startup message information.
+type startupMessageEntry struct {
+	key   string
+	value string
+}
+
+// ListenData contains the listener metadata provided to OnListenHandler.
 type ListenData struct {
-	Host string
-	Port string
-	TLS  bool
+	ColorScheme Colors
+	Host        string
+	Port        string
+	Version     string
+	AppName     string
+
+	ChildPIDs []int
+
+	HandlerCount int
+	ProcessCount int
+	PID          int
+
+	TLS     bool
+	Prefork bool
+}
+
+// PreStartupMessageData contains metadata exposed to OnPreStartupMessage hooks.
+type PreStartupMessageData struct {
+	PrimaryInfo   Map
+	SecondaryInfo Map
+
+	Header string
+
+	ColorScheme Colors
+	Host        string
+	Port        string
+	Version     string
+	AppName     string
+
+	ChildPIDs []int
+
+	HandlerCount int
+	ProcessCount int
+	PID          int
+
+	TLS            bool
```

Comment:
Maybe this immutable data can be embedded into the struct using ListenData struct. They don't seem to have an impact on startup message when they change